app/components/provider_interface/status_box_components/offer_component.rb

```diff
@@ -54,11 +54,11 @@ def change_path(target)
 
         case target
         when :provider
-          paths.provider_interface_application_choice_change_offer_edit_provider_path(application_choice.id, entry: 'provider') if show_provider_link?
+          paths.provider_interface_application_choice_edit_offer_path(application_choice.id, step: 'provider') if show_provider_link?
```

Comment:
There's no need for a case statement here, you can simplify as:

```ruby
paths.provider_interface_application_choice_edit_offer_path(application_choice.id, step: target.to_s)
```

You could check that `target` is valid beforehand if need be.

Reply:
Normally I would agree but there is some logic in `show_provider_link?`, `show_course_link?` and `show_course_option_link?` that needs to be checked according to `target`. This is probably the main reason for the current construct.

Reply:
Ah yes didn't spot that link check. 
You could use the target to work out the collection needed to calculate whether the link was displayed. Then you'd just have a method called `show_link?(target)`. This would lead to less less `show_*_link?` methods and you wouldn't need this case statement:

```ruby
def show_link?(target)
  collection = case target
  when :course then available_course
  when :course_options then available_course_options
  else available_providers
  end
  collection.count > 1 if collection
end
```

Reply:
I like